androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/fetchApi.kt

```diff
@@ -0,0 +1,33 @@
+package com.mbta.tid.mbta_app.android.util
+
+import com.mbta.tid.mbta_app.model.response.ApiResult
+import com.mbta.tid.mbta_app.repositories.IErrorBannerStateRepository
+
+/**
+ * If `getData` returns an `ApiResultOk`, clears any preexisting data error in `errorKey` in
+ * `errorBannerRepo` and calls `onSuccess`. If `getData` returns an `ApiResultError` or throws, sets
+ * an error in `errorKey` in `errorBannerRepo` with the given `onRefreshAfterError`.
+ */
+suspend fun <T : Any> fetchApi(
+    errorBannerRepo: IErrorBannerStateRepository,
+    errorKey: String,
+    getData: suspend () -> ApiResult<T>,
+    onSuccess: suspend (T) -> Unit = {},
+    onRefreshAfterError: () -> Unit
+) {
+    val result: ApiResult<T> =
+        try {
+            getData()
+        } catch (e: Exception) {
+            ApiResult.Error(code = null, message = e.message ?: "No error message")
```

Comment:
suggestion: I _think_ "No error message" would only be exposed in dev debug mode, but can leave it empty string to prevent unlocalized string from surfacing in UI